src/fontra/views/editor/panel-transformation.js

```diff
@@ -451,6 +530,77 @@ export default class TransformationPanel extends Panel {
     };
   }
 
+  async doPathOperations(pathOperationFunc, undoLabel) {
+    let { point: pointIndices } = parseSelection(this.sceneController.selection);
+    pointIndices = pointIndices || [];
+
+    const positionedGlyph =
+      this.sceneController.sceneModel.getSelectedPositionedGlyph();
+
+    const allContourIndices = [...range(positionedGlyph.glyph.path.numContours)];
+    const selectedContourIndices =
+      !pointIndices.length && undoLabel === "Remove overlaps"
+        ? allContourIndices
+        : [...getSelectionByContour(positionedGlyph.glyph.path, pointIndices).keys()];
+
+    if (!selectedContourIndices.length) {
+      return;
+    }
+
+    const notSelectedContourIndices = allContourIndices.filter(
+      (n) => !selectedContourIndices.includes(n)
+    );
+    if (!notSelectedContourIndices.length && undoLabel !== "Remove overlaps") {
+      return;
+    }
+
+    const editLayerGlyphs = this.sceneController.getEditingLayerFromGlyphLayers(
+      positionedGlyph.varGlyph.glyph.layers
+    );
+
+    const layerPaths = await mapObjectValuesAsync(
+      editLayerGlyphs,
+      async (layerGlyph) => {
+        const path = layerGlyph.path;
+        const selectedContoursPath = new VarPackedPath();
+        const unselectedContoursPath = new VarPackedPath();
+
+        for (const contourIndex of range(path.numContours)) {
+          if (selectedContourIndices.includes(contourIndex)) {
+            selectedContoursPath.appendContour(path.getContour(contourIndex));
+          } else {
+            unselectedContoursPath.appendContour(path.getContour(contourIndex));
+          }
+        }
+
+        return await pathOperationFunc(selectedContoursPath, unselectedContoursPath);
+      }
+    );
+
+    await this.sceneController.editGlyphAndRecordChanges(
+      (glyph) => {
+        for (const [layerName, newPath] of Object.entries(layerPaths)) {
+          const path = glyph.layers[layerName].glyph.path;
```

Comment:
I would restructure:
```js
  const glyph = glyph.layers[layerName]
  if (doUnion) {
    ...delete selected contours...
    append new path to glyph.path
  } else {
    glyph.path = ...thenewpath...  // just replace the whole thing
  }
```

Reply:
I used `glyph.layers[layerName].glyph.path = layerPath;` but it fails and I don't know why. I compared the code you did for the knife tool, but I cannot see what causes it to fail.

The contours outlines are gone, but displayed in light grey and an error pops up in the console:

<img width="1816" alt="Screenshot 2024-08-01 at 15 18 08" src="https://github.com/user-attachments/assets/7c882f43-f71d-4a53-b733-f4cb63e49d7f">

Reply:
Please see: https://github.com/googlefonts/fontra/pull/1582/commits/0ae336495c127ddbf38df68d6d3494a001b19406

Reply:
Fixed it: 
[Bugfix: layerPath was not a VarPackedPath](https://github.com/googlefonts/fontra/pull/1582/commits/c363c2084cb87a05864abe94ce2e3bbd1e0a5d64)